### PR TITLE
fix(auth): handle false token expiration dates and ensure boolean return

### LINKED ISSUE #841: Refactor: Improve type safety and edge-case handling in isTokenExpired (off.ts)
### What
The `isTokenExpired` function in `off.ts` handles falsy values incorrectly because it depends on the logical AND (`&&`) operator (`return payload.exp && Date.now() >= payload.exp * 1000;`). This leads to two edge cases:
1. **The `exp = 0` Issue:** Tokens that the backend revokes with `exp = 0` return `0` (falsy) instead of `true`, causing the system to incorrectly consider the expired token as valid.
2. **The `undefined` Issue:** Tokens without an `exp` claim return `undefined` instead of a strict boolean. This breaks type safety and strict equality checks (for example, `=== false`).

### Steps to reproduce the behavior:
1. Pass a JWT to `OpenFoodFacts.isTokenExpired()` where the decoded payload includes `"exp": 0` (this simulates a token that has been revoked immediately).
2. Check the function's return value.
3. You will see an error: The function returns `0` (which is falsy) instead of evaluating the timestamp and returning `true` (expired).
4. *Alternatively:* Pass a token without an `exp` claim, and you will see the function returns `undefined` instead of a boolean.

### Expected behavior
The function should ensure it always returns a strict `boolean`. If `exp = 0`, it should correctly evaluate the timestamp and return `true` (expired). If the `exp` claim is missing, it should handle it properly (for example, returning `false`) without producing `undefined`.

### Screenshots
*(Not applicable - Code/Logic issue)*

### Platform (Desktop, Mobile, Hunger Games)
 - OS: All
 - Platform: JavaScript/TypeScript SDK

### Part of
- Related to the ongoing cleanup of auth header/token handling seen in #834

### PR DESCRIPTION
**Pull request by saishmungase**
### What
This PR rewrites the JWT expiration logic in `OpenFoodFacts.isTokenExpired()` (`off.ts`). It checks for `null` or `undefined` directly instead of relying on truthiness.

Previously, `payload.exp && Date.now() >= payload.exp * 1000` had two edge cases:
1. **The `exp = 0` Issue:** Tokens revoked by the backend with an `exp` of `0` were treated as falsy. This bypassed the timestamp check and returned `0`.
2. **Type Safety:** Tokens lacking an `exp` claim returned `undefined`, which is not a strict `boolean`.

This update ensures a strict `: boolean` return type and treats `exp = 0` as an expired token.

**Testing Added:** I have also added dedicated test cases in `tests/off.spec.ts` to ensure that `exp = 0` and missing `exp` claims are evaluated correctly, preventing future regressions.

### Screenshot
N/A - Internal logic and type-safety fix.

### Fixes bug(s)
- Fixes #841 

### Part of 
- N/A (Standalone tech-debt cleanup)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved token expiration handling so tokens missing an expiry are treated as valid and truly expired tokens are correctly recognized, reducing unexpected sign-outs and improving authentication stability.

* **Tests**
  * Added tests covering edge cases around token expiry claims to prevent regressions and ensure consistent authentication behavior.

### CODE REVIEW

File: src/off.ts [hunk 1]
```diff
@@ -304,17 +304,20 @@ export class OpenFoodFacts {
...
       throw new Error("Invalid JWT token format");
     }
     const payload = JSON.parse(
       Buffer.from(parts[1], "base64").toString("utf-8"),
-    ) as { exp: number };
+    ) as { exp?: number };
 
+    if (payload.exp == null) {
+      return false; 
+    }
```
Comment (VaiTon) [inline src/off.ts]: We can remove this and add it back to the final return

```
return payload.exp != null && ...
```

**Review by VaiTon — APPROVED** [review]
LGTM
